packages/interactivity/src/test/utils.ts

```diff
@@ -25,4 +27,256 @@ describe( 'Interactivity API', () => {
+				steps.push( { scope: getScope(), namespace: getNamespace() } );
+				const b = yield Promise.resolve( a + 1 );
+				steps.push( { scope: getScope(), namespace: getNamespace() } );
+				return b * 2;
+			}
+			const callback = dummyScopeAndNS( () => withScope( gen ) );
+			const result = await callback();
+
+			expect( result ).toBe( 4 );
+			steps.forEach( ( step ) => {
+				expect( step.scope ).toBe( dummyScope );
+				expect( step.namespace ).toBe( dummyNamespace );
+			} );
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should yield correct values when promises resolve in generator functions', async () => {
+			function* gen() {
+				const a = yield Promise.resolve( 3 );
+				const b = yield Promise.resolve( a + 2 );
+				return b;
+			}
+			const wrapped = dummyScopeAndNS( () => withScope( gen ) );
+			const result = await wrapped();
+			expect( result ).toBe( 5 );
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should return the resolved value when a promise is returned in generator functions', async () => {
+			function* gen() {
+				const a = yield Promise.resolve( 3 );
+				return Promise.resolve( a + 2 );
+			}
+			const wrapped = dummyScopeAndNS( () => withScope( gen ) );
+			const result = await wrapped();
+			expect( result ).toBe( 5 );
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should accept arguments in generator functions like in normal functions', async () => {
+			function* gen( ...values: number[] ) {
+				let result = 0;
+				for ( const value of values ) {
+					result += yield Promise.resolve( value );
+				}
+				return result;
+			}
+			const wrapped = dummyScopeAndNS( () => withScope( gen ) );
+			const result = await wrapped( 1, 2, 3, 4 );
+			expect( result ).toBe( 10 );
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should propagate errors thrown inside a generator function', async () => {
+			function* gen() {
+				throw new Error( 'Test Error' );
+			}
+			const wrapped = dummyScopeAndNS( () => withScope( gen ) );
+			await expect( wrapped() ).rejects.toThrow( 'Test Error' );
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should propagate errors when a generator function returns a rejected promise', async () => {
+			function* gen() {
+				return Promise.reject( new Error( 'Test Error' ) );
+			}
+			const wrapped = dummyScopeAndNS( () => withScope( gen ) );
+			await expect( wrapped() ).rejects.toThrow( 'Test Error' );
+			expect( getScope() ).toBe( previousScope );
+			expect( getNamespace() ).toBe( previousNamespace );
+		} );
+
+		it( 'should handle captured errors within generator execution and resume correctly', async () => {
```

Comment:
This test is redundant with the following one (`should maintaining scope and namespace when a generator captures an error `).